http4k-jsonrpc/src/main/kotlin/org/http4k/jsonrpc/ErrorMessage.kt

```diff
@@ -0,0 +1,29 @@
+package org.http4k.jsonrpc
+
+import org.http4k.format.Json
+
+open class ErrorMessage(val code: Int, val message: String): JsonNodeProducer {
+
+    open fun <ROOT : NODE, NODE> data(json: Json<ROOT, NODE>): NODE? = null
+
+    final override fun <ROOT : NODE, NODE> invoke(json: Json<ROOT, NODE>): NODE {
+        val fields = listOf(
+                "code" to json.number(code),
+                "message" to json.string(message)
+        )
+        val data = data(json)
+        return json.obj(data?.let { fields + ("data" to it) } ?: fields)
+    }
+
+    companion object {
+        val InvalidRequest = ErrorMessage(-32600, "Invalid Request")
+        val MethodNotFound = ErrorMessage(-32601, "Method not found")
+        val InvalidParams = ErrorMessage(-32602, "Invalid params")
+        val InternalError = ErrorMessage(-32603, "Internal error")
+        val ParseError = ErrorMessage(-32700, "Parse error")
+    }
+}
+
+interface JsonNodeProducer {
```

Comment:
Is this interface needed? looks like there is only one usage

Reply:
No I guess not. Could simply be incorporated in to the ErrorMessage class. I was going to use that interface in other domain classes, but didn't really plan out that way.

Reply:
I've removed it anyway in tweaks I've made to the code already - so might be worth you rebasing from master to get those. :)

Reply:
Couple of other things. There are a few unused methods in extensionsKt. Are these required - or at least maybe meant to be covered by a test?

Also, for some reason coveralls is complaining that there is 0% coverage on the new module, which is a bit annoying as it's dragging down the percentage score. :(  If you could have a look to see what might be missing that'd be ace.

Reply:
Np
 - I'm away now as well! 

Take a look at the other branch. And let me know what you think. I made some tweaks to hide the construction of the builder.

Reply:
Ah ok. I'll have a more detailed look and let you know what I think. Initial feedback is that it look ok to me. The reason I chose to use extension methods was that I wanted to base it on what was done for the Contract module. Using methods on an object might make more sense though and makes it more explicit.

Reply:
LGTM. Created another PR for some minor changes to that branch: https://github.com/http4k/http4k/pull/174